src/main/java/janggi/domain/piece/Piece.java

```diff
@@ -0,0 +1,47 @@
+package janggi.domain.piece;
+
+import janggi.domain.board.Position;
+
+import java.util.List;
+
+public abstract class Piece {
+    final PieceColor color;
+    final PieceType type;
+
+    Piece(PieceColor color, PieceType type) {
+        this.color = color;
+        this.type = type;
+    }
+
+    int countPieceInRoute(List<Piece> piecesInRoute) {
+        return (int) piecesInRoute.stream()
+                .filter(piece -> piece.color != PieceColor.NONE)
```

Comment:
NONE 직접 접근해서 사용하고 있네요

```suggestion
                .filter(piece -> piece.color.isNotNone())
```

Reply:
Piece 자체가 Empty임을 확인하는 메서드를 만들어 사용하도록 변경했습니다!

```java
// Piece.java
public abstract boolean isNotEmptyPiece();
```